.github/workflows/on_build.yml

```diff
@@ -7,19 +7,25 @@ on:
       - master
 
 jobs:
-  build:
+  build-gradle-project:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
+      - name: Checkout project sources
         uses: actions/checkout@v4
 
       - name: Set up JDK 17
         uses: actions/setup-java@v4
         with:
           java-version: '17'
           distribution: 'temurin'
-          cache: 'maven'
 
-      - name: Build with Maven
-        run: mvn --batch-mode -P github package
+      - name: Cache Gradle packages
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: "${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}"
 
```

Comment:
When using caching in GA, remember to set restore-keys.
Ref. -> https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action